tests/test_attributes.py

```diff
@@ -0,0 +1,257 @@
+                {% load component_tags %}
+                <div {{ component_vars.attrs }} >
+                </div>
+            """
+
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component "test" attrs:@click.stop="dispatch('click_event')" attrs:x-data="{hello: 'world'}" attrs:class=class_var %}
+            {% endcomponent %}
+        """
+        template = Template(template_str)
+        rendered = template.render(Context({"class_var": "padding-top-8"}))
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <div @click.stop="dispatch('click_event')" x-data="{hello: 'world'}" class="padding-top-8" >
+            </div>
+            """,
+        )
+
+    def test_attrs_accessible_in_get_context_data(self):
+        @component.register("test")
+        class AttrsComponent(component.Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <div {{ attrs }} >
+                </div>
+            """
+
+            def get_context_data(self, *args, attrs):
+                return {"attrs": attrs}
+
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component "test" attrs:@click.stop="dispatch('click_event')" attrs:x-data="{hello: 'world'}" attrs:class=class_var %}
+            {% endcomponent %}
+        """
+        template = Template(template_str)
+        rendered = template.render(Context({"class_var": "padding-top-8"}))
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <div @click.stop="dispatch('click_event')" x-data="{hello: 'world'}" class="padding-top-8" >
+            </div>
+            """,
+        )
+
+    def test_attrs_immutable(self):
+        @component.register("test")
+        class AttrsComponent(component.Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <div {{ attrs }} >
+                </div>
+            """
+
+            def get_context_data(self, *args, attrs):
+                attrs["my_super_key"] = "abc"
+                return {"attrs": attrs}
+
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component "test" attrs:@click.stop="dispatch('click_event')" attrs:x-data="{hello: 'world'}" attrs:class=class_var %}
+            {% endcomponent %}
+        """
+        template = Template(template_str)
+        with self.assertRaisesMessage(TypeError, "cannot change object - object is immutable"):
+            template.render(Context({"class_var": "padding-top-8"}))
+
+    def test_attrs_tag(self):
+        @component.register("test")
+        class AttrsComponent(component.Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <div {% merge_attrs component_vars.attrs class+="added_class" data-id=123 %}>
```

Comment:
So there's 2 ways to render the attributes into the template. If user does not need to change the attributes, they can simply call `{{ component_vars.attrs }}` as seen here https://github.com/EmilStenstrom/django-components/pull/491/files#diff-a22fb83b06ea7d4165ec648dddac689cf529335fc2dc994c61eb1250ca26465cR216

But if they need to add extra fields along with the attrs, then, I used the same approach for merging/rendering the attributes as used in [`django-web-components`](https://github.com/Xzya/django-web-components/tree/master?tab=readme-ov-file#passing-data-to-components). That is, using a tag to render the dict and specify any additional fields as kwargs on the tag.

Reply:
I'm not a big fan of adding the "+=" operator here. My long term hope is to merge this into the django codebase, and each new thing like this increases the things you have to understand, and that need to be "sold" to the django maintainers.

For this, I'd suggest we drop the "+" sign, and just do "=" to mean "add".

Reply:
Maybe we could similar thing as we did with the `attrs`, and have something like `add:class="abc"`, so there's still a way to differentiate between "adding" and "setting" attributes, but without the unconventional `+=`?